### PR TITLE
feat(helm):Allow remote values.yaml with -f

### LINKED ISSUE #2642: Allow remote values.yaml with -f
It would be great to be able to use `-f` option with remote files, same as `kubectl apply -f` works.  

``` sh
helm install my-repo/my-service-template
  -f https://config-repo.example.com/my-namespace/my-service-config.yml
```

### PR DESCRIPTION
**Pull request by rocky-nupt**
In this feature, we can use -f option with remote files, same as kubectl
accepted URLs. I add an option to send a 'get' request when read the local
file failed.

Closes #2642

### CODE REVIEW

**Comment by rimusz** [comment]
nice one

File: cmd/helm/install.go
Comment (nebril) [inline cmd/helm/install.go]: I think this line isn't needed anymore.

File: cmd/helm/install.go
Comment (nebril) [inline cmd/helm/install.go]: This will cause non-existent files to return error `Get /path/to/filename: unsupported protocol scheme ""` instead of `open /path/to/filename: no such file or directory`, which IMO is a regression in user experience.

I think you need to differentiate between filename and url, the simplest way you could implement it would be by using `net/url.Parse`. You can check if the string is http url if it parses and has http/https scheme.

**Review by nebril — CHANGES_REQUESTED** [review]
Thanks for the PR, I think it provides good value, please refer to my comments inline.

File: cmd/helm/install.go
Comment (rocky-nupt) [inline cmd/helm/install.go]: hah, yes. I forgot it. Thank you for your remind.:)

File: cmd/helm/install.go
Comment (rocky-nupt) [inline cmd/helm/install.go]: I have just learned GO for nearly  a month. There is too much I do not know. Your suggestion is so useful. Thank you for your help!

File: cmd/helm/install.go
Comment (nebril) [inline cmd/helm/install.go]: Small nitpick: you can just `return ioutil.ReadFile(filePath)`

File: cmd/helm/install.go
Comment (nebril) [inline cmd/helm/install.go]: I wonder if we want to handle other protocols (ftp anybody still uses that? :D) as well, but I guess it should be filed as a separate issue.

**Review by nebril — CHANGES_REQUESTED** [review]
Looks better, please fix error handling part.

**Comment by technosophos** [comment]
To stay consistent with the rest of Helm, this should go through the protocol resolution library just like `helm fetch` and other URL resolvers.

You can start tracing here to see how we resolve and fetch URLs: https://github.com/kubernetes/helm/blob/master/cmd/helm/fetch.go#L119

File: cmd/helm/install.go
Comment (rocky-nupt) [inline cmd/helm/install.go]: Yes. You are right. Maybe someone handle other protocols. I can leave a FIXME here.

File: cmd/helm/install.go
Comment (rocky-nupt) [inline cmd/helm/install.go]: Oh, I ignored this. Thank you for your help!

File: cmd/helm/install.go
Comment (rocky-nupt) [inline cmd/helm/install.go]: Thx!

**Comment by rocky-nupt** [comment]
I have followed your suggestion and changed it. Please review it again. :) @technosophos

**Comment by innovia** [comment]
This is for fetching values for a remote url, what about installing from remote with using a values.yaml ?

I might be missing something but now I'm doing
```bash
helm fetch stable/something 
helm install .something -f something_values.yaml
```
I would like it to be something like this:
``` helm install stable/something -f my_values.yaml ```

**Comment by rocky-nupt** [comment]
But where we fetch the remote values.yaml? @innovia

**Comment by bacongobbler** [comment]
The remote chart is tarball'd so you'll have to unpack it separately from the chart if you want to install it from the remote chart, @innovia.

```
><> curl -s https://kubernetes-charts.storage.googleapis.com/index.yaml | grep wordpress.*.tgz | head -n 1 | awk '{ print $2 }'
https://kubernetes-charts.storage.googleapis.com/wordpress-0.6.10.tgz
```

You could write a plugin that fetches the chart and uses the values.yaml within the chart if you wanted to. You won't be able to do that given the current helm repo architecture unless you served those files outside of the chart.

**Comment by rocky-nupt** [comment]
Your idea is great. But I think you can create a new issue. I am glad to finish it if you allow.:）

**Comment by innovia** [comment]
Ok i'll open up a new issue thanks for the clarification

**Comment by rocky-nupt** [comment]
Excuse me,  @nebril . Can you review this PR again？Thx！

**Comment by technosophos** [comment]
Thanks for making the changes. This looks good to me, but I'll give @nebril the final word since he's the official reviewer.

**Review by nebril — APPROVED** [review]
LGTM, sorry for a long wait!

**Comment by thomastaylor312** [comment]
@nebril Feel free to merge this when ready. We are getting ready for 2.7

**Comment by very-doge-wow** [comment]
A quick question, is there any way I can configure helm to use credentials for authentication at the registry where the remote values file is stored? I'm trying to use this in an enterprise context where everything is locked behind basic auth...

**Pull request merged**
